src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,215 @@
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT cars.id AS car_id, cars.model AS model, "
+                + "manufacturers.id AS manufacturer_id, "
+                + "manufacturers.country AS country, manufacturers.name manufacturer_name "
+                + "FROM cars INNER JOIN manufacturers "
+                + "ON manufacturers.id = cars.manufacturer_id "
+                + "WHERE cars.is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(setCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars from DB.", e);
+        }
+        for (int i = 0; i < cars.size(); i++) {
+            cars.get(i).setDrivers(getAllByCar(cars.get(i).getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, "
+                + "manufacturer_id = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update car " + car, e);
+        }
+        deleteDriversFromCar(car.getId());
+        for (int i = 0; i < car.getDrivers().size(); i++) {
+            addDriverToCar(car.getDrivers().get(i).getId(), car.getId());
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        deleteDriversFromCar(id);
+        String queryDeleteCar = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                  PreparedStatement statement = connection.prepareStatement(queryDeleteCar)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT DISTINCT cars.id AS car_id, cars.model AS model, "
+                + "manufacturer_id, "
+                + "manufacturers.name AS manufacturer_name, "
+                + "manufacturers.country AS country, "
+                + "drivers.id AS driver_id, drivers.name AS driver_name, "
+                + "drivers.license_number AS license_number "
```

Comment:
```suggestion
        String query = "SELECT DISTINCT cars.id AS car_id, cars.model AS model, "
                + "manufacturer_id, "
                + "manufacturers.name AS manufacturer_name, "
                + "manufacturers.country AS country, "
```

нам не потрібна тут інформація про водія, ти її ніде не використовуєш

Reply:
Yap, done